compiler warning in test_MKLDNN and etc.

test_MKLDNN.cpp:
```
[07:16:03]W:	 [Step 1/1] /paddle/paddle/gserver/tests/test_MKLDNN.cpp:300:3:   required from here
[07:16:03]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:721:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[07:16:03]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_GE, >=)
[07:16:03]W:	 [Step 1/1]                                 ^
[07:16:03]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:148:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[07:16:03]W:	 [Step 1/1]  #define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[07:16:03]W:	 [Step 1/1]                                                      ^
[07:16:03]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:721:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[07:16:03]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_GE, >=)
```
ROIPoolLayer.cpp:
```
[06:53:39]W:	 [Step 1/1] In file included from /paddle/build/third_party/install/protobuf/include/google/protobuf/stubs/common.h:40:0,
[06:53:39]W:	 [Step 1/1]                  from /paddle/build/proto/ModelConfig.pb.h:9,
[06:53:39]W:	 [Step 1/1]                  from /paddle/paddle/gserver/layers/Layer.h:19,
[06:53:39]W:	 [Step 1/1]                  from /paddle/paddle/gserver/layers/ROIPoolLayer.h:17,
[06:53:39]W:	 [Step 1/1]                  from /paddle/paddle/gserver/layers/ROIPoolLayer.cpp:15:
[06:53:39]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h: In instantiation of 'std::__cxx11::string* google::Check_GEImpl(const T1&, const T2&, const char*) [with T1 = long unsigned int; T2 = int; std::__cxx11::string = std::__cxx11::basic_string<char>]':
[06:53:39]W:	 [Step 1/1] /paddle/paddle/gserver/layers/ROIPoolLayer.cpp:101:5:   required from here
[06:53:39]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:721:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[06:53:39]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_GE, >=)
```

--- a/paddle/gserver/layers/ROIPoolLayer.cpp
+++ b/paddle/gserver/layers/ROIPoolLayer.cpp
@@ -98,7 +98,7 @@ void ROIPoolLayer::forward(PassType passType) {
     size_t roiStartH = round(bottomROIs[2] * spatialScale_);
     size_t roiEndW = round(bottomROIs[3] * spatialScale_);
     size_t roiEndH = round(bottomROIs[4] * spatialScale_);
-    CHECK_GE(roiBatchIdx, 0);
+    CHECK_GE(roiBatchIdx, 0UL);
     CHECK_LT(roiBatchIdx, batchSize);
     size_t roiHeight = std::max(roiEndH - roiStartH + 1, 1UL);
     size_t roiWidth = std::max(roiEndW - roiStartW + 1, 1UL);

--- a/paddle/gserver/tests/test_MKLDNN.cpp
+++ b/paddle/gserver/tests/test_MKLDNN.cpp
@@ -297,7 +297,7 @@ static void getAddtoConfig(TestConfig& cfg,
 }
 
 void testAddtoLayer(const testImageDesc& pm, const size_t nInputs) {
-  CHECK_GE(nInputs, 1);
+  CHECK_GE(nInputs, 1UL);
   TestConfig dnnConfig;
   getAddtoConfig(dnnConfig, pm, nInputs);
   dnnConfig.layerConfig.set_type("mkldnn_addto");

